src/dataflow/src/sink/kafka.rs

```diff
@@ -211,15 +221,13 @@ impl ProducerContext for SinkProducerContext {
     fn delivery(&self, result: &DeliveryResult, _: Self::DeliveryOpaque) {
         match result {
             Ok(_) => (),
-            Err((e, msg)) => {
+            Err((_e, msg)) => {
                 self.metrics.message_delivery_errors_counter.inc();
-                error!(
-                    "received error while writing to kafka sink topic {}: {}",
-                    msg.topic(),
-                    e
-                );
+                // Should be bounded by the size of the kafka buffer
+                self.retry_queue.lock().unwrap().push_back(msg.detach());
```

Comment:
We want to get the retry semantics that `KafkaSinkState::send` provides.  The `rdkafka` queue could be full, for example, or we could hit a fatal error we want to shutdown because of